pkg/admission/prep_downscale.go

```diff
@@ -460,6 +464,41 @@ func createEndpoints(ar v1.AdmissionReview, oldInfo, newInfo *objectInfo, port,
 	return eps
 }
 
+func invokePrepareShutdown(ctx context.Context, method string, parentLogger log.Logger, client httpClient, ep endpoint) error {
+	logger, ctx := spanlogger.New(ctx, parentLogger, "admission.PreparePodForShutdown", tenantResolver)
+	defer logger.Span.Finish()
+
+	logger.SetSpanAndLogTag("url", ep.url)
+	logger.SetSpanAndLogTag("index", ep.index)
+
+	req, err := http.NewRequestWithContext(ctx, method, "http://"+ep.url, nil)
+	if err != nil {
+		level.Error(logger).Log("msg", fmt.Sprintf("error creating HTTP %s request", method), "err", err)
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req, ht := nethttp.TraceRequest(opentracing.GlobalTracer(), req)
+	defer ht.Finish()
+
+	resp, err := client.Do(req)
+	if err != nil {
+		level.Error(logger).Log("msg", fmt.Sprintf("error sending HTTP %s request", method), "err", err)
+		return err
```

Comment:
this line was absent from this routine before. I'm 90% sure that was a bug.

Reply:
👍  Ah yeah, it was the one after http.NewRequest, not client.Do.

https://github.com/grafana/rollout-operator/blob/ea171936c5266041fda49506bac3aa2aa23e075e/pkg/admission/prep_downscale.go#L481-L484